torch/csrc/jit/codegen/cuda/test/test_gpu_view.cpp

```diff
@@ -1341,6 +1342,607 @@ TEST_F(NVFuserTest, FusionReductionFlatten1_CUDA) {
+  auto tv6 = makeConcreteTensor({a, b, c, 1, 1, 1});
+  fusion.addInput(tv6);
+
+  // c, e, and f are tied together this should be counted as a broadcast dim in
+  // the reference since it's a partial bcast
+  auto tv7 = makeConcreteTensor({a, b, 1, 1, 1, 1});
+  fusion.addInput(tv7);
+
+  // plug the broadcasts into the fusion
+  auto tv8 = add(tv5, tv4);
+  auto tv9 = add(tv6, tv8);
+  auto tv10 = add(tv7, tv9);
+  fusion.addOutput(tv10);
+
+  auto bcast_info =
+      scheduler_utils::getBroadcastMultiples(tv4, DataType::Int32);
+
+  // linked c, e, and f together so they should have the same id.
+  TORCH_CHECK(bcast_info.view_disjoint_set_ids[5] == 0);
+  TORCH_CHECK(bcast_info.view_disjoint_set_ids[4] == 0);
+  TORCH_CHECK(bcast_info.view_disjoint_set_ids[3] == 1);
+  TORCH_CHECK(bcast_info.view_disjoint_set_ids[2] == 0);
+  TORCH_CHECK(bcast_info.view_disjoint_set_ids[1] == 2);
+  TORCH_CHECK(bcast_info.view_disjoint_set_ids[0] == 3);
+
+  TORCH_CHECK(
+      scheduler_utils::breakIsDisjoint(bcast_info.view_disjoint_set_ids, 0));
+  TORCH_CHECK(
+      scheduler_utils::breakIsDisjoint(bcast_info.view_disjoint_set_ids, 1));
+  TORCH_CHECK(
+      scheduler_utils::breakIsDisjoint(bcast_info.view_disjoint_set_ids, 2));
+  TORCH_CHECK(
+      !scheduler_utils::breakIsDisjoint(bcast_info.view_disjoint_set_ids, 3));
+  TORCH_CHECK(
+      !scheduler_utils::breakIsDisjoint(bcast_info.view_disjoint_set_ids, 4));
+  TORCH_CHECK(
+      !scheduler_utils::breakIsDisjoint(bcast_info.view_disjoint_set_ids, 5));
+
+  // tv0  [a, b, c, d, e, f]
+  // tv1  [a, b, c, d, e, f]
+  // tv3  [a, b, c, d, e, f]
+  // tv4  [a, b, c, d, e, f]
+  // tv5  [1, 1, c, 1, e, f] -> Left bcasts should show up in some multiples
+  // tv6  [a, b, c, 1, 1, 1] -> view interferes with bcasts, non of these should
+  //                            show up
+  // tv7  [a, b, 1, 1, 1, 1] -> These broadcasts could be recognized
+  // tv10 [a, b, c, d, e, f]
+
+  TORCH_CHECK(
+      bcast_info.broadcast_multiples[0].lhs_multiple == 0 &&
+      bcast_info.broadcast_multiples[0].rhs_multiple == 8 * 4);
+
+  TORCH_CHECK(
+      bcast_info.broadcast_multiples[1].lhs_multiple == 7 * 4 &&
+      bcast_info.broadcast_multiples[1].rhs_multiple == 8 * 4);
+
+  TORCH_CHECK(
+      bcast_info.broadcast_multiples[2].lhs_multiple == 7 * 4 &&
+      bcast_info.broadcast_multiples[2].rhs_multiple == 7 * 4);
+
+  TORCH_CHECK(
+      bcast_info.broadcast_multiples[3].lhs_multiple == 8 * 4 &&
+      bcast_info.broadcast_multiples[3].rhs_multiple == 7 * 4);
+
+  TORCH_CHECK(
+      bcast_info.broadcast_multiples[4].lhs_multiple == 8 * 4 &&
+      bcast_info.broadcast_multiples[4].rhs_multiple == 7 * 4);
+
+  TORCH_CHECK(
+      bcast_info.broadcast_multiples[5].lhs_multiple == 8 * 4 &&
+      bcast_info.broadcast_multiples[5].rhs_multiple == 7 * 4);
+}
+
+// Make sure matching views are segmented into the same kernel
+TEST_F(NVFuserTest, FusionViewMagicSchedule1_CUDA) {
```

Comment:
CC @jjsjann123 and @rdspring1 please try to think of any other view/alias tests we can add (can be in a follow up).